### PR TITLE
openblas: Enable riscv64-linux

### PR DESCRIPTION
**Pull request by zhaofengli**
###### Motivation for this change

Well, guess I'm too late to add this to the same eval as #122035.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126393` at 3cc4d9bc run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>47 packages marked as broken and skipped:</summary>
  <ul>
    <li>blastem</li>
    <li>cassandra_2_1</li>
    <li>cassandra_2_2</li>
    <li>curaByDagoma</li>
    <li>cura_stable</li>
    <li>displaycal</li>
    <li>gnuradio3_7</li>
    <li>gnuradio3_7Packages.ais</li>
    <li>gnuradio3_7Packages.gnuradio</li>
    <li>gnuradio3_7Packages.gsm</li>
    <li>gnuradio3_7Packages.limesdr</li>
    <li>gnuradio3_7Packages.nacl</li>
    <li>gnuradio3_7Packages.osmosdr</li>
    <li>gnuradio3_7Packages.rds</li>
    <li>googleearth-pro</li>
    <li>grass</li>
    <li>hadoop</li>
    <li>hadoop_2_7</li>
    <li>hadoop_2_8</li>
    <li>hadoop_2_9</li>
    <li>hplipWithPlugin_3_16_11</li>
    <li>hplipWithPlugin_3_18_5</li>
    <li>hplip_3_16_11</li>
    <li>hplip_3_18_5</li>
    <li>impressive</li>
    <li>invoice2data</li>
    <li>julia</li>
    <li>julia-lts</li>
    <li>julia_10</li>
    <li>linuxPackages_hardkernel_4_14.virtualbox</li>
    <li>linuxPackages_hardkernel_latest.virtualbox</li>
    <li>loxodo</li>
    <li>metamorphose2</li>
    <li>ocropus</li>
    <li>odpdown</li>
    <li>opentsdb</li>
    <li>pdfread</li>
    <li>pig</li>
    <li>plover.stable</li>
    <li>rainbowstream</li>
    <li>scribus</li>
    <li>scyther</li>
    <li>spark</li>
    <li>termplay</li>
    <li>torchat</li>
    <li>winpdb</li>
    <li>yaxg</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/e4eac6ac73d46284267c879422600602">python38Packages.brotlipy</a></li>
    <li><a href="https://gist.github.com/6c3b2f3bd2c4bb69e60730eb4ac81ebc">python38Packages.construct</a></li>
    <li><a href="https://gist.github.com/dcec89149cbac84331ae4da7d0736e53">python38Packages.httpbin</a></li>
    <li><a href="https://gist.github.com/187db4a3bbe70e3002d3b4f2b156759c">python38Packages.pytest-httpbin</a></li>
    <li><a href="https://gist.github.com/143dc1ac33f919469865a6bf70959987">python38Packages.pytest-vcr</a></li>
    <li><a href="https://gist.github.com/5c53404c992234776b52081f95d1ea2d">python38Packages.vcrpy</a></li>
  </ul>
</details>
<details>
  <summary>3243 packages skipped due to time constraints:</summary>
  <ul>
    <li>DisnixWebService</li>
    <li>MACS2</li>
    <li>MIDIVisualizer</li>
    <li>R</li>
    <li>abcl</li>
    <li>actdiag (python38Packages.actdiag)</li>
    <li>adapta-gtk-theme</li>
    <li>adoptopenjdk-icedtea-web</li>
    <li>aiodnsbrute</li>
    <li>alloy (alloy5)</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>31 packages built successfully:</summary>
  <ul>
    <li>arpack</li>
    <li>auditBlasHook</li>
    <li>blas</li>
    <li>csdp</li>
    <li>lapack</li>
    <li>openblas</li>
    <li>openblasCompat</li>
    <li>python38Packages.bokeh</li>
    <li>python38Packages.boto</li>
    <li>python38Packages.bottleneck</li>
    <li>python38Packages.emcee</li>
    <li>python38Packages.fastprogress</li>
    <li>python38Packages.httplib2</li>
    <li>python38Packages.httpretty</li>
    <li>python38Packages.imagecodecs-lite</li>
    <li>python38Packages.nose-randomly</li>
    <li>python38Packages.numexpr</li>
    <li>python38Packages.numpy</li>
    <li>python38Packages.oauth2client</li>
    <li>python38Packages.pillow</li>
    <li>python38Packages.pytest-arraydiff</li>
    <li>python38Packages.pytest-doctestplus</li>
    <li>python38Packages.pytest-filter-subpackage</li>
    <li>python38Packages.pytest-randomly</li>
    <li>python38Packages.pytools</li>
    <li>python38Packages.reportlab</li>
    <li>python38Packages.spglib</li>
    <li>python38Packages.tifffile</li>
    <li>python38Packages.uncertainties</li>
    <li>qrupdate</li>
    <li>suitesparse (suitesparse_5_3)</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/development/libraries/science/math/openblas/default.nix [hunk 1]
```diff
@@ -133,6 +140,15 @@ stdenv.mkDerivation rec {
     sha256 = "1qjr02cqncv20abdp1yzr55n7smhx6h9chqvb0xbp18byynvj87w";
   };
 
+  # remove this when updating to 0.3.16
+  patches = [
+    (fetchpatch {
+      name = "riscv64-fixes.patch";
+      url = "https://github.com/xianyi/OpenBLAS/pull/3259.patch";
```
Comment (mweinelt) [inline pkgs/development/libraries/science/math/openblas/default.nix:147]: Can we fetch by referencing the exact commit ids instead? Pull requests are tracking a branch, that could, in theory, still move.

**Comment by mweinelt** [comment]
You dropped 590be3fae35d134fae156c60dc3a21d7933f8914, is that on purpose?

**Comment by zhaofengli** [comment]
> You dropped 590be3fae35d134fae156c60dc3a21d7933f8914, is that on purpose?

Yes, because that's not necessary for a successful build. I added the Makefile just to fix the top-level `clean` target.

**Review by mweinelt — APPROVED** [review]

**Pull request merged**